example/src/lib.rs

```diff
@@ -0,0 +1,128 @@
+mod state;
+
+use fil_actors_runtime::runtime::{ActorCode, Runtime};
+use fil_actors_runtime::{actor_error, ActorDowncast, ActorError, cbor};
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::RawBytes;
+use fvm_shared::error::ExitCode;
+use fvm_shared::{MethodNum, METHOD_CONSTRUCTOR};
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
+use crate::state::{State, UserPersistParam};
+
+/// SCA actor methods available
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    /// Constructor for Storage Power Actor
+    Constructor = METHOD_CONSTRUCTOR,
+    Persist = 2,
+}
+
+/// Subnet Coordinator Actor
+pub struct Actor;
+impl Actor {
+    /// Constructor for SCA actor
+    fn constructor<BS, RT>(rt: &mut RT) -> Result<(), ActorError>
+        where
+            BS: Blockstore,
+            RT: Runtime<BS>,
+    {
+        let st = State::new(rt.store()).map_err(|e| {
```

Comment:
You can check here if the constructor call is from the INIT_ACTOR , as a sanity-check for the constructor